csharp/src/Microsoft.ML.OnnxRuntime/OrtLoraAdapter.shared.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    /// Represents Lora Adapter in memory
+    /// </summary>
+    public class OrtLoraAdapter : SafeHandle
+    {
+        /// <summary>
+        /// Creates an instance of OrtLoraAdapter from file.
+        /// The adapter file is memory mapped. If allocator parameter
+        /// is provided, then lora parameters are copied to the memory
+        /// allocated by the specified allocator.
+        /// </summary>
+        /// <param name="adapterPath">path to the adapter file</param>
+        /// <param name="ortAllocator">optional allocator, can be null, must be a device allocator</param>
+        /// <returns>New instance of LoraAdapter</returns>
+        public static OrtLoraAdapter Create(string adapterPath, OrtAllocator ortAllocator)
+        {
+            var platformPath = NativeOnnxValueHelper.GetPlatformSerializedString(adapterPath);
+            var allocatorHandle = (ortAllocator != null) ? ortAllocator.Pointer : IntPtr.Zero;
+            NativeApiStatus.VerifySuccess(NativeMethods.CreateLoraAdapter(platformPath, allocatorHandle,
+                out IntPtr adapterHandle));
+            return new OrtLoraAdapter(adapterHandle);
+        }
+
+        /// <summary>
+        /// Creates an instance of OrtLoraAdapter from an array of bytes.
```

Comment:
Would be good to point out that the bytes are copied into the new OrtLoraAdapter so the user knows they don't need to remain valid.